### PR TITLE
fix(Core/Groups): fix group enchanting level not resetting properly

### PR DESCRIPTION
**Pull request by r-o-b-o-t-o**
## Changes Proposed:
-  Call `ResetMaxEnchantingLevel()` after removing the member from the party slot, so that the computed enchanting level takes into account the member leaving

## Issues Addressed:
- Closes https://github.com/chromiecraft/chromiecraft/issues/96

## Tests Performed:

- Applied the C++ fix
- Tested in-game on Windows 10 clients and server

## How to Test the Changes:

Setup:
Get three characters
- Character A: without the enchanting skill
- Character B: without the enchanting skill
- Character C: **with** the enchanting profession

First step:
- Get characters A, B and C into a group
- Move all characters to Dr. Whitherlimb (`.go c 1975927`)
- Kill Dr. Whitherlimb
- He should loot a green item, verify that you are able to disenchant it:
image

Second step:
- Remove all three characters from the group
- Get characters A, B and C into a group
- With character C, leave the group
- Do `.respawn` on the Dr.
- Kill Dr. Whitherlimb
- He should loot a green item, verify that you are **NOT** able to disenchant it:
image

Third step: (thanks @eTzmNcbkrng for pointing this out)
- Remove all characters from the group
- Get characters A, B and C into a group
- Disconnect character C from the game without removing them from the group (this can be achieved for example by using the `.kick` command in the server console):
image
- Do `.respawn` on the Dr.
- Kill Dr. Whitherlimb
- He should loot a green item, verify that you are **NOT** able to disenchant it:
image

(if this is not clear, see my comment and video on how to reproduce the bug on the issue: https://github.com/chromiecraft/chromiecraft/issues/96#issuecomment-827807897)

## Known Issues and TODO List:

None

## Target Branch(es):
- [x] Master

 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by eTzmNcbkrng** [comment]
Sorry, I'm unable to test this currently, does this fix also correctly reevaluate if the enchanter character is offline but not removed from the group?

File: src/server/game/Groups/Group.cpp [hunk 1]
```diff
@@ -602,6 +598,10 @@ bool Group::RemoveMember(ObjectGuid guid, const RemoveMethod& method /*= GROUP_R
                 sWorld->UpdateGlobalPlayerGroup(guid.GetCounter(), 0);
         }
 
+        // Reevaluate group enchanter if the leaving player had enchanting skill or the player is offline
+        if (!player || player->GetSkillValue(SKILL_ENCHANTING))
+            ResetMaxEnchantingLevel();
```
Comment (FrancescoBorzi) [inline src/server/game/Groups/Group.cpp:603]:
```suggestion
        if (!player || player->GetSkillValue(SKILL_ENCHANTING))
        {
            ResetMaxEnchantingLevel();
        }
```

**Comment by r-o-b-o-t-o** [comment]
> Sorry, I'm unable to test this currently, does this fix also correctly reevaluate if the enchanter character is offline but not removed from the group?

Hi @eTzmNcbkrng, good catch! It did not refresh when a character went offline indeed, should be fixed now 👍🏻

**Comment by Sedatyf** [comment]
Hey guys!
Here is my tests:
- First Step ✔️ 

https://user-images.githubusercontent.com/59012307/116589678-2da70b80-a91d-11eb-9b3f-87b4e946d351.mp4

- Second Step ✔️ 

https://user-images.githubusercontent.com/59012307/116589711-37307380-a91d-11eb-994c-71be12e5ddda.mp4

- Third Step ✔️ 

https://user-images.githubusercontent.com/59012307/116589742-3ef01800-a91d-11eb-8440-2be2015d24a8.mp4

Thank you @r-o-b-o-t-o for giving us Test Cases, pretty useful. Thanks @eTzmNcbkrng for pointing this case. Good thinking!
It just need a review 😄

**Review by FrancescoBorzi — APPROVED** [review]

**Review by Helias — APPROVED** [review]

**Pull request merged**